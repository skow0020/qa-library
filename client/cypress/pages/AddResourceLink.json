{
    "title": "#title",
    "category": "#category",
    "language": "#language",
    "url": "#url",
    "backgroundImage": "#backgroundImage",
    "description": "#body"
}