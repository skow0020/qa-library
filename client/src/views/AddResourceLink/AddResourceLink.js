import AlertModal, { showAlert } from "components/common/AlertModal";
import React, { useState } from "react";

import Button from '@material-ui/core/Button';
import CategoriesSelection from "components/common/CategoriesSelection";
import Colors from 'utils/Colors';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import LanguagesSelection from "components/common/LanguagesSelection";
import PageTitle from "components/common/PageTitle";
import TextField from "components/common/TextField";
import axios from "axios";
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  submit: {
    margin: theme.spacing(3, 0, 2),
    backgroundColor: Colors.primary,
    color: Colors.white,
    marginLeft: "auto"
  },
  paper: {
    marginTop: theme.spacing(2),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  form: {
    width: '100%',
    marginTop: theme.spacing(3),
    justifyContent: 'center'
  },
  textField: {
    backgroundColor: Colors.white
  }
}));

export default function AddBook(props) {
  const classes = useStyles();
  const [backgroundImage, setBackgroundImage] = useState('');
  const [category, setCategory] = useState('General');
  const [url, setUrl] = useState('');
  const [title, setTitle] = useState('');
  const [body, setBody] = useState('');
  const [language, setLanguage] = useState('');

  const handleSubmit = (event) => {
    event.preventDefault();
    const resourceLink = {
      backgroundImage,
      category,
      url,
      title,
      setBody,
      language
    };

    axios.post("/api/resourceLinks/", resourceLink)
      .then(
        response => {
          if (response.data.success) showAlert({ message: "ResourceLink added successfully" });
          else showAlert({ message: `Unable to add resource link: ${JSON.stringify(response.data.error.errors)}` });
          nextPath('/resource-links');
        },
        error => showAlert({ message: `Unable to add resource link: ${error}` })
      );
  };

  const nextPath = (path) => {
    setTimeout(() => {
      props.history.push(path);
    }, 2000);
  };

  return (
    <Container component="main">
      <AlertModal />
      <div className={classes.paper}>
        <form id="add-article-form" className={classes.form} onSubmit={handleSubmit}>
          <Grid container>
            <PageTitle title="Add a Resource Link" />
          </Grid>
          <Grid container spacing={2}>
            <Grid item xs={12} lg={12} md={12} sm={12}>
              <TextField id="title" label="Title" value={title} onChange={(e) => setTitle(e.target.value)} required />
            </Grid>
            <Grid item lg={3} md={6}>
              <LanguagesSelection id="langauge" value={language} onChange={(e) => setLanguage(e.target.value)} />
            </Grid>
            <Grid item lg={3} md={6}>
              <CategoriesSelection id="category" value={category} onChange={(e) => setCategory(e.target.value)} />
            </Grid>
            <Grid item xs={12} sm={6} md={12}>
              <TextField id="url" label="URL" value={url} onChange={(e) => setUrl(e.target.value)} required />
            </Grid>
            <Grid item xs={12} sm={6} md={12}>
              <TextField id="backgroundImage" label="Background Image" value={backgroundImage} onChange={(e) => setBackgroundImage(e.target.value)} required />
            </Grid>
            <Grid item xs={12} sm={12}>
              <TextField id="body" label="Description" value={body} multiline={true} rows={2} rowsMax={4} onChange={(e) => setBody(e.target.value)} />
            </Grid>
          </Grid>
          <Button type="submit" variant="contained" id="submit-button" className={classes.submit} fullWidth>
            Submit
          </Button>
        </form>
      </div>
    </Container >
  );
}
