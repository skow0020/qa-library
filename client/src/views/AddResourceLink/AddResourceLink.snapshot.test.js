import AddResourceLink from './AddResourceLink';
import React from 'react';
import renderer from 'react-test-renderer';

describe('AddResourceLink Snapshot Tests', () => {
  test('AddResourceLink snapshot', () => {
    const component = renderer.create(
      <AddResourceLink></AddResourceLink>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});