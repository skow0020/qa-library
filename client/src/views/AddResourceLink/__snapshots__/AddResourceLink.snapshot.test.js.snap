// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddResourceLink Snapshot Tests AddResourceLink snapshot 1`] = `
<div
  className="main-content-container px-4 container-fluid"
>
  <div />
  <form
    className=""
    id="add-book-form"
    onSubmit={[Function]}
  >
    <div
      className="page-header py-4 no-gutters row"
    >
      <div
        className="text-sm-left text-center text-md-left mb-sm-0 col-12 col-sm-8"
      >
        <span
          className="text-uppercase page-subtitle"
        />
        <h3
          className="page-title"
        >
          Add a Resource Link
        </h3>
      </div>
      <button
        className="btn btn-success btn-lg btn btn-primary"
        id="submit"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="title"
    >
      Resource Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <div
      className="form-row"
    >
      <div
        className="col"
      >
        <label
          className="text-muted font-weight-bold px-2"
          htmlFor="language"
        >
          Languages
        </label>
        <select
          className="form-control custom-select"
          id="language"
          onChange={[Function]}
          type="text"
          value=""
        >
          <option />
          <option>
            Java
          </option>
          <option>
            CSharp
          </option>
          <option>
            Python
          </option>
          <option>
            Ruby
          </option>
          <option>
            JavaScript
          </option>
          <option>
            Cpp
          </option>
          <option>
            Swift
          </option>
        </select>
      </div>
      <div
        className="col"
      >
        <label
          className="text-muted font-weight-bold px-2"
          htmlFor="category"
        >
          Categories
        </label>
        <select
          className="form-control custom-select"
          id="category"
          onChange={[Function]}
          required={true}
          type="text"
          value="General"
        >
          <option />
          <option>
            General
          </option>
          <option>
            UI Automation
          </option>
          <option>
            API Automation
          </option>
          <option>
            Unit Testing
          </option>
          <option>
            DevOps
          </option>
          <option>
            Development
          </option>
          <option>
            Databases
          </option>
        </select>
      </div>
    </div>
    <label
      className="text-muted font-weight-bold"
      htmlFor="url"
    >
      Resource URL
    </label>
    <input
      className="form-control"
      id="url"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="backgroundImage"
    >
      Background Image URL
    </label>
    <input
      className="form-control"
      id="backgroundImage"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="text-muted font-weight-bold"
      htmlFor="body"
    >
      Resource Description
    </label>
    <textarea
      className="form-control"
      id="body"
      onChange={[Function]}
      style={
        Object {
          "height": "150px",
        }
      }
      type="textarea"
      value=""
    />
  </form>
</div>
`;
